e2e/lookout_ingester_test/lookout_ingester_test.go

```diff
@@ -0,0 +1,200 @@
+package lookout_ingester_test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/G-Research/armada/internal/common/util"
+	"github.com/G-Research/armada/internal/lookout/repository"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+var defaultWaits struct {
+	PollWait     time.Duration
+	ExpiresAfter time.Duration
+}
+
+func init() {
+	pollWait, _ := time.ParseDuration("1s")
+	expiresAfter, _ := time.ParseDuration("1m")
+
+	defaultWaits.PollWait = pollWait
+	defaultWaits.ExpiresAfter = expiresAfter
+}
+
+func TestLookoutIngesterUpdatesPostgresWithJobInfo(t *testing.T) {
+	err := client.WithConnection(connectionDetails(), func(connection *grpc.ClientConn) error {
+		submitClient := api.NewSubmitClient(connection)
+		jobRequest := createJobRequest("personal-anonymous")
+
+		createQueue(submitClient, jobRequest, t)
+
+		submitResponse, err := client.SubmitJobs(submitClient, jobRequest)
+		assert.NoError(t, err)
+		assert.Equal(t, len(submitResponse.JobResponseItems), 1)
+
+		jobId := submitResponse.JobResponseItems[0].JobId
+		errStr := submitResponse.JobResponseItems[0].Error
+		assert.Empty(t, errStr)
+		assert.NotEmpty(t, jobId)
+
+		db, err := openPgDbTestConnection()
+		assert.NoError(t, err)
+
+		pJobRun, err := awaitJobRunEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJobRun)
+		assert.Equal(t, pJobRun.JobId, submitResponse.JobResponseItems[0].JobId)
+
+		pJob, err := awaitJobEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter, repository.JobSucceededOrdinal)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJob)
+		assert.Equal(t, pJob.JobId, submitResponse.JobResponseItems[0].JobId)
+
+		return nil
+	})
+	assert.NoError(t, err)
+}
+
+func TestLookoutIngesterUpdatesPostgresWithJobInfoFailedJob(t *testing.T) {
```

Comment:
I like this but note there's a couple of different ways that a job can fail and each way determines what is written into lookout.  Specifically if a job actually starts up (which it won't in this case) we have to write the exit code(s) into the db.